src/SearchParameters/index.js

```diff
@@ -1265,16 +1264,12 @@ SearchParameters.prototype = {
    * @return {string[]}
    */
   getRefinedHierarchicalFacets: function getRefinedHierarchicalFacets() {
-    var hierarchicalRefinedFacets = Object.keys(
-      this.hierarchicalFacetsRefinements
+    return intersection(
+      // enforce the order between the two arrays,
```

Comment:
I'm not completely sure to understand the comment. What does it mean?

Reply:
This comment was there originally by the way

Reply:
Ah. It doesn't make sense to me anymore.

Reply:
I think we can remove this comment here

Reply:
The output is an array that contains elements from both passed arrays, so we might wonder which of the two arrays dictates the order.

In the case of [lodash intersection function](https://lodash.com/docs/4.17.11#intersection), it's the first array. 
Example:
```js
intersection([1, 2, 3, 4, 5], [5, 3]) === [3, 5] 
intersection([1, 2, 3, 4, 5], [3, 5]) === [3, 5]
intersection([1, 2, 5, 4, 3], [3, 5]) === [5, 3]
```


Whoever used intersection here wanted to keep the order of `this.hierarchicalFacets.map(...)` over the the order of `Object.keys(this.hierarchicalFacetsRefinements)`.